cli/src/delete_snapshots_command.py

```diff
@@ -0,0 +1,143 @@
+ID of a snapshot.
+"""
+
+INCLUDE_INACTIVE_HELP = """
+If set, also delete snapshots which have been completed. By default, only
+pending snapshots will be deleted. This flag is not required when specifying the
+exact ID of an inactive snapshot.
+"""
+
+QUIET_HELP = 'If set, suppresses user confirmation of the command.'
+
+SUMMARY_HEADERS = ['Status', 'Location', 'Condition', 'ID']
+
+
+def transform_to_snapshot_summary(snapshot):
+  # Match the fields from SUMMARY_HEADERS
+  return [
+      'COMPLETED' if snapshot['isFinalState'] else 'ACTIVE',
+      breakpoint_utils.transform_location_to_file_line(snapshot['location']),
+      snapshot['condition'] if 'condition' in snapshot else '', snapshot['id']
+  ]
+
+
+class DeleteSnapshotsCommand:
+  """This class implements the delete_snapshots command.
+
+  The register() method is called by the CLI startup code to install the
+  delete_snapshots command information, and the cmd() function will be invoked
+  if the delete_snapshots command was specified by the user.
+  """
+
+  def __init__(self):
+    pass
+
+  def register(self, args_subparsers, required_parsers, common_parsers):
+    parent_parsers = [
+        common_parsers.database_url, common_parsers.format,
+        common_parsers.use_default_rtdb, common_parsers.debuggee_id
+    ]
+    parent_parsers += required_parsers
+    parser = args_subparsers.add_parser(
+        'delete_snapshots', description=DESCRIPTION, parents=parent_parsers)
+    parser.add_argument('ID', help=ID_HELP, nargs='*')
+    parser.add_argument('--all-users', help=ALL_USERS_HELP, action='store_true')
+    parser.add_argument(
+        '--include-inactive', help=INCLUDE_INACTIVE_HELP, action='store_true')
+    parser.add_argument('--quiet', help=QUIET_HELP, action='store_true')
+    parser.set_defaults(func=self.cmd)
+
+  def cmd(self, args, cli_services):
+    user_input = cli_services.user_input
+    user_output = cli_services.user_output
+    firebase_rtdb_service = cli_services.get_firebase_rtdb_service()
+
+    command_utils.validate_debuggee_id(user_output, firebase_rtdb_service,
+                                       args.debuggee_id)
+
+    # This will be a list, if no IDs were specified it will be empty. If any IDs
+    # are specified those are the only ones that will be deleted.
+    snapshot_ids = args.ID
+
+    user_email = None if args.all_users is True else cli_services.account
+
+    snapshots = []
+
+    if snapshot_ids:
+      ids_not_found = []
+
+      for bp_id in snapshot_ids:
+        snapshot = breakpoint_utils.get_breakpoint(firebase_rtdb_service,
+                                                   args.debuggee_id, bp_id)
+        if snapshot is None:
+          ids_not_found.append(bp_id)
+        else:
+          snapshots.append(snapshot)
+
+      if ids_not_found:
+        user_output.error(f"Snapshot ID not found: {', '.join(ids_not_found)}")
+        raise SilentlyExitError
```

Comment:
Nit: it probably makes sense to allow the user to continue even with some snapshots not found.  The scenario that motivates this thinking for me is a user attempting to retry a partially completed deletion.

Reply:
Created https://github.com/GoogleCloudPlatform/snapshot-debugger/issues/10 for this.